[network.interface_ip] Correct handling of interfaces without IP

kbespalov opened the pull request:
**Issue:**
If an interface have no any assigned IP address, then the func fails with the following traceback:
```
root@cfg01:/home/ubuntu# salt-call network.interface_ip eth10 --local
[ERROR   ] An un-handled exception was caught by salt's global exception handler:
KeyError: 0
Traceback (most recent call last):
  File "/usr/bin/salt-call", line 11, in <module>
    salt_call()
  File "/usr/lib/python2.7/dist-packages/salt/scripts.py", line 372, in salt_call
    client.run()
  File "/usr/lib/python2.7/dist-packages/salt/cli/call.py", line 58, in run
    caller.run()
  File "/usr/lib/python2.7/dist-packages/salt/cli/caller.py", line 134, in run
    ret = self.call()
  File "/usr/lib/python2.7/dist-packages/salt/cli/caller.py", line 197, in call
    ret['return'] = func(*args, **kwargs)
  File "/usr/lib/python2.7/dist-packages/salt/modules/network.py", line 792, in interface_ip
    return salt.utils.network.interface_ip(iface)
  File "/usr/lib/python2.7/dist-packages/salt/utils/network.py", line 847, in interface_ip
    return iface_info.get(iface, {}).get('inet', {})[0].get('address', '')
KeyError: 0

```

**Expected result:**
 - the call must returns empty string if there is no IP assigned to network interface.

**Steps to reproduce:**

1. Create a veth pair as example of net interfaces without IP:
```
ip link add veth0 type veth peer name veth1
```

2. Try to achive IP using `network.interface_ip` module:
```
salt-call network.interface_ip  veth0 --local
```

salt-jenkins commented:
@kbespalov, thanks for your PR! By analyzing the history of the files in this pull request, we identified @jfindlay, @thatch45 and @mirceaulinic to be potential reviewers.

kbespalov committed:
[network.interface_ip] Fix handling of interfaces without IP

If an interface have no any assigned IP address, then the func fails
with the following:

  root@cfg01:/home/ubuntu# salt-call network.interface_ip eth10 --local
  [ERROR   ] An un-handled exception was caught by salt's global exception handler:
  KeyError: 0

Expected result:
 - the call must returns empty string if there is no IP of an interface.

KumarRajnish commented:
All,

Due to a misconfiguration on our Office365 tenant, all email addressed to your Ruckus Wireless email address (you@ruckuswireless.com) were delivered to a catch all mailbox.  This incident occurred between 11 PM to 8 AM (PST).  IT is delivering these emails with a “Reply-all” rule to all individuals on the email chain.  If you feel you are missing email attachments, please send email to helpdesk@ruckuswireless.com.

We apologize for any inconvenience this may have caused.

Regards,

IT

From: kbespalov [mailto:notifications@github.com]
Sent: Friday, September 15, 2017 8:24 AM
To: saltstack/salt <salt@noreply.github.com>
Cc: Subscribed <subscribed@noreply.github.com>
Subject: [saltstack/salt] [network.interface_ip] Correct handling of interfaces without IP (#43524)


Issue:
If an interface have no any assigned IP address, then the func fails with the following traceback:

root@cfg01:/home/ubuntu# salt-call network.interface_ip eth10 --local

[ERROR   ] An un-handled exception was caught by salt's global exception handler:

KeyError: 0

Traceback (most recent call last):

  File "/usr/bin/salt-call", line 11, in <module>

    salt_call()

  File "/usr/lib/python2.7/dist-packages/salt/scripts.py", line 372, in salt_call

    client.run()

  File "/usr/lib/python2.7/dist-packages/salt/cli/call.py", line 58, in run

    caller.run()

  File "/usr/lib/python2.7/dist-packages/salt/cli/caller.py", line 134, in run

    ret = self.call()

  File "/usr/lib/python2.7/dist-packages/salt/cli/caller.py", line 197, in call

    ret['return'] = func(*args, **kwargs)

  File "/usr/lib/python2.7/dist-packages/salt/modules/network.py", line 792, in interface_ip

    return salt.utils.network.interface_ip(iface)

  File "/usr/lib/python2.7/dist-packages/salt/utils/network.py", line 847, in interface_ip

    return iface_info.get(iface, {}).get('inet', {})[0].get('address', '')

KeyError: 0



Expected result:

  *   the call must returns empty string if there is no IP assigned to network interface.

________________________________
You can view, comment on, or merge this pull request online at:

  https://github.com/saltstack/salt/pull/43524

Commit Summary

  *   [network.interface_ip] Fix handling of interfaces without IP

File Changes

  *   M salt/utils/network.py<https://github.com/saltstack/salt/pull/43524/files#diff-0> (3)

Patch Links:

  *   https://github.com/saltstack/salt/pull/43524.patch
  *   https://github.com/saltstack/salt/pull/43524.diff

—
You are receiving this because you are subscribed to this thread.
Reply to this email directly, view it on GitHub<https://github.com/saltstack/salt/pull/43524>, or mute the thread<https://github.com/notifications/unsubscribe-auth/AHusM9hDgTsSzzUJf8csK1f1DbOQjdMcks5sipaMgaJpZM4PZIwl>.

KumarRajnish commented:
All,

Due to a misconfiguration on our Office365 tenant, all email addressed to your Ruckus Wireless email address (you@ruckuswireless.com) were delivered to a catch all mailbox.  This incident occurred between 11 PM to 8 AM (PST).  IT is delivering these emails with a “Reply-all” rule to all individuals on the email chain.  If you feel you are missing email attachments, please send email to helpdesk@ruckuswireless.com.

We apologize for any inconvenience this may have caused.

Regards,

IT

From: SaltStack Jenkins [mailto:notifications@github.com]
Sent: Friday, September 15, 2017 8:24 AM
To: saltstack/salt <salt@noreply.github.com>
Cc: Subscribed <subscribed@noreply.github.com>
Subject: Re: [saltstack/salt] [network.interface_ip] Correct handling of interfaces without IP (#43524)


@kbespalov<https://github.com/kbespalov>, thanks for your PR! By analyzing the history of the files in this pull request, we identified @jfindlay<https://github.com/jfindlay>, @thatch45<https://github.com/thatch45> and @mirceaulinic<https://github.com/mirceaulinic> to be potential reviewers.

—
You are receiving this because you are subscribed to this thread.
Reply to this email directly, view it on GitHub<https://github.com/saltstack/salt/pull/43524#issuecomment-329814658>, or mute the thread<https://github.com/notifications/unsubscribe-auth/AHusM_q4w7GUcsJMB_zKfJ_OlzcFXvdHks5sipaYgaJpZM4PZIwl>.

rallytime commented:
re-run lint

rallytime commented:
re-run docs
